homeassistant/components/sensor/wunderground.py

```diff
@@ -639,13 +644,11 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     for variable in config[CONF_MONITORED_CONDITIONS]:
         sensors.append(WUndergroundSensor(hass, rest, variable))
 
-    rest.update()
+    yield from rest.async_update()
```

Comment:
If you do this update before initializing the sensors, you will not do unnecessary work. It also means that you don't have to pass `True` to `async_add_devices` because the sensors have access to data the moment it is initialized.

Reply:
> If you do this update before initializing the sensors, you will not do unnecessary work.

With the current design of this platform (which isn't ideal, imho), when a sensor entity is created it tells the data holder that its "feature" should be monitored, so doing `WUndergroundData#async_update()` before the sensors have added their features would result in an empty query. We could however instead just pass the features into the data holder directly and not through the sensors, but I tried to limit this PR for now as this platform has many improvements that could be made (using the dispatcher to not rely on polling, ...)

> It also means that you don't have to pass `True` to `async_add_devices`

The reason I'm passing `True` to `async_add_devices` is that the **only way** for sensors to get new data is through their own `WUndergroundSensor#async_update`. They don't get notified by `WUndergroundData` that a request has been made. Instead, they periodically **poll** data from the data holder, which also takes care of throttling the request. Therefore the only way to have *some* data in the entities before `async_update` is automatically called is by passing `True`  to `async_add_devices`. Again, this could be fixed by completely reworking this platform.

Reply:
okay, fine fine 👍

Reply:
I'm sorry if this was perceived as being a bit rude - that was not my intention. I wanted to explain why I did this so that you, and anybody attempting to change this part of the codebase in the future, can follow what's going on. It's always better to ask when some code snippet "smells wrong" 👍.

Anyway, the underlying issue of this component being weird with it using polling still persists. But I think changing that would make the most sense during a transition to the `weather` component (btw, I'm not using this platform too much, so I wouldn't be the one who would like to do that 🤓).